### PR TITLE
Fix fseek bug on SD card

### PR DESCRIPTION
**Pull request by stefanmielke**
This PR fixes a bug on `debug.c` that would cause `fseek` to not return its current poisition, causing a bug when trying to get the size of a file on the SD card, as verified by the test changed to check for the behavior (fails before the fix).

It also adds an example that uses the SD card to load and display sprites (that only works with this fix).

### CODE REVIEW

File: examples/loadspritefromsd/loadspritefromsd.c [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+		debugf("Error loading background\n");
+		return NULL;
+	}
+
+	const int size = filesize(fp);
+	debugf("Size: %d\n", size);
+
+	sprite_t *sprite = malloc(size);
+
+	setvbuf(fp, NULL, _IONBF, 0);
```
Comment (rasky) [inline examples/loadspritefromsd/loadspritefromsd.c]: Do you really need this?

File: examples/loadspritefromsd/loadspritefromsd.c [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+		sprite = spritesheet_load(sprites_sd[cur_sprite]);
+	else
+		sprite = spritesheet_load(sprites_rom[cur_sprite]);
+}
+
+int main(void) {
+	/* Initialize peripherals */
+	display_init(RESOLUTION_320x240, DEPTH_16_BPP, 2, GAMMA_NONE, ANTIALIAS_RESAMPLE);
+	dfs_init(DFS_DEFAULT_LOCATION);
+	rdp_init();
```
Comment (rasky) [inline examples/loadspritefromsd/loadspritefromsd.c]: This is probably not needed

File: examples/loadspritefromsd/loadspritefromsd.c [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	controller_init();
+	timer_init();
```
Comment (rasky) [inline examples/loadspritefromsd/loadspritefromsd.c]: Also this

**Review by rasky — APPROVED** [review]
Thanks!

File: examples/loadspritefromsd/loadspritefromsd.c (hunk 1 above)
Comment (stefanmielke) [reply examples/loadspritefromsd/loadspritefromsd.c]: Nope, removed (along with the other unused code).

**Pull request merged**